### DISCUSSION TITLE
[atom]  Why is there no method getIn?

### DISCUSSION THREAD

**Discussion by bit-app-3000** [thi-ng/umbrella#487] [Q&A] [answered]
Greetings

Why is there no method   getIn ? 

I am looking for a simple solution to get the value in Atom by  path

```
import { defAtom } from '@thi.ng/atom'

let seed = { key1: { label: 'one' }}

let db = defAtom(seed)

let label = db.getIn(['key1', 'label'])

```

**Comment by postspectacular** [comment, marked as answer]
Hi @bit-app-3000 — apologies I forgot to respond to this, in case you're still needing an answer: The deep accessor functionality like `getIn`, `setIn`, `updateIn` was alwyas meant to be kept a separate feature (so it can also be used with other data structures, incl. vanilla JS objects) and therefore is available as separate package: [thi.ng/paths](https://thi.ng/paths). In my experience I found updating nested values in an atom is much more common than reading nested values directly. For the latter it's much more common to have (reactive) [views](https://docs.thi.ng/umbrella/atom/index.html#md:derived-views) or [cursors](https://docs.thi.ng/umbrella/atom/index.html#md:cursor) defined, and so atoms only provide nested-value update functions like `resetIn`, `updateIn`, but not nested readonly accessors like `getIn`... so far no one else has brought this up as a shortcoming 🤷‍♂️ (but I'm happy to accept a PR :)

So to use `getIn` with an atom you can do like so:

```ts
import { defAtom } from "@thi.ng/atom";
import { getIn, getInUnsafe } from "@thi.ng/paths";

const db = defAtom({ a: { b: { c: 42 } } });

// access via type checked path
console.log(getIn(db.deref(), ["a","b","c"]));

// or unchecked path
console.log(getInUnsafe(db.deref(), ["a","b","c"]));

// or path as string
console.log(getInUnsafe(db.deref(), "a.b.c"));
```
